consensus/beacon/oplegacy.go

```diff
@@ -0,0 +1,69 @@
+package beacon
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rpc"
+	"math/big"
+)
+
+type OpLegacy struct {
+}
+
+func (o *OpLegacy) Author(header *types.Header) (common.Address, error) {
+	return header.Coinbase, nil
+}
+
+func (o *OpLegacy) VerifyHeader(chain consensus.ChainHeaderReader, header *types.Header) error {
+	return nil // legacy chain is verified by block-hash reverse sync
```

Comment:
Is this because of asking for blocks by hash? I can't find anything that explicitly checks the parent hash relationship in the headerchain / verify header code

Reply:
The skeleton chain mechanism should be checking the parent-hash matches the parent block. The more-recent header that we can check the `hash(header) == recent.blockhash` of isn't here in `VerifyHeader`.

And since this is the `OpLegacy` validation, it should be used as part of the regular downloader / upon getting data from the skeleton chain. 

In the beacon variant of the consensus code there is:
```
	parent := chain.GetHeader(header.ParentHash, header.Number.Uint64()-1)
	if parent == nil {
		return consensus.ErrUnknownAncestor
	}
```
The docs of the skeleton downloader state that the headers are stored "ephemerally until sync finishes".
But since this code uses the headers DB I think it's only called in the regular downloader / after prior headers are accepted into the DB already, which means skeleton sync results are there.

So I think we may want to add this header check to ensure DB consistency, but the parent-hash checking is otherwise taken care of by the skeleton sync. Specifically the sub-chains linking/merging is what should ensure the header parent-hash consistency.

Reply:
Added this check. Tracking for `OpLegacy.VerifyHeaders` in https://github.com/ethereum-optimism/protocol-quest/issues/90